src/animation/friction.ts

```diff
@@ -0,0 +1,79 @@
+import { rAF } from '../rAF';
+import { Entity, applyForce } from '../forces/Force.gen';
+import { frictionForceV } from '../forces/Friction.gen';
+import { VectorSetter } from './types';
+
+export interface Friction1DParams {
+  config: {
+    mu: number;
+    normal?: number;
+    velocity: number;
+  };
+  onUpdate: VectorSetter;
+  onComplete: () => void;
+}
+
+type FrictionState = Entity;
+
+const applyForceForStep = (
+  entity: FrictionState,
+  mu: number,
+  normal?: number
+) => {
+  const force = frictionForceV({ mu, normal, velocity: entity.velocity });
+
+  return applyForce({ force, entity });
+};
+
+export const friction1D = (params: Friction1DParams) => {
+  let state: FrictionState = {
+    mass: 1,
```

Comment:
Since we're making the normal force a scalar constant, `mass` can just be 1 here. On a 2d surface we'd derive the normal force using the angle between the direction of the normal force and the gravitational force. But for the purpose of these 1d animations, we're assuming a flat surface where the normal force and gravitational force cancel one another out.